### PR TITLE
feature: add the ability to associate a registration to a user

### PR DESCRIPTION
**Pull request by ColinW520**
Registrations, in this system, serve as a data container pertinent to the individual registrant. Given the importance of the fact that each individual registrant must **SIGN THEIR OWN WAIVER**, we can use the registration class to be the association with the individual user (member, in this parlance), in pursuit of a user profile experience.

The current registration experience **DOES NOT** support the creation of accounts. This PR introduces the bare minimum of the functionality needed to support that.

To test:

 - [x] Pull down this branch
 - [x] Run migrations 
 - [x] Sign in as an admin
 - [x] Register for an available race

^ everything should work, and the user_id value should be set on your registration

### CODE REVIEW

File: app/views/registrations/_form.html.haml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 = simple_form_for([@event, @registration], validate: true) do |f|
   = f.input :started_at, as: :hidden, input_html: { value: @registration.started_at }
   = f.input :step_to_validate, as: :hidden, input_html: { value: "start" }
+  = f.input, :user_id, as: :hidden, input_html: { value: current_user.try(:id) }
```
Comment (ColinW520) [inline app/views/registrations/_form.html.haml]: @willblake01 -- this is importat. I suggest doing some reading on how `try` works

File: app/models/registration.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ class Registration < ApplicationRecord
 
   belongs_to :event
   belongs_to :discount_code, counter_cache: true, optional: true
+  belongs_to :user, optional: true
```
Comment (ColinW520) [inline app/models/registration.rb]: @willblake01 -- this is important. I recommend doing some googling on why the `optional` param is necessary here.

File: app/views/registrations/_form.html.haml (hunk 1 above)
Comment (willblake01) [inline app/views/registrations/_form.html.haml]: Looks good! I was getting a syntax error on this line for the comma after `= f.input`. Did that make its way in on accident?

Comment (willblake01) [reply app/views/registrations/_form.html.haml]: Will do!

File: app/models/registration.rb (hunk 2 above)
Comment (willblake01) [reply app/models/registration.rb]: 👍

**Comment by ColinW520** [comment]
> Just wanted to make sure the comma in this line `= f.input, :user_id, as: :hidden, input_html: { value: current_user.try(:id) }` doesn't need to be removed before merging.

Great catch! I will get that change made.

**Pull request merged**